crates/forge_analyzer/src/definitions.rs

```diff
@@ -226,9 +233,96 @@ pub fn run_resolver(
         module.visit_with(&mut collector);
     }
 
+    // This loop iterates through env's bodies and recreates an updated set of bodies to satisfy SSA form of IR dump.
+    let mut vars_map: HashMap<VarId, (VarKind, Option<VarId>, bool)> = HashMap::new();
+    for (func_id, body) in environment.bodies_mut().enumerate() {
+        // Builds the variables map to track existing all variables, store their VarId, and check if they have been assigned once (by SSA).
+        for var_id in body.vars.keys() {
+            let var_kind = *body.vars.get(var_id).unwrap();
+            vars_map.insert(var_id, (var_kind, None, false));
+        }
+        // Updates the instruction set with creation of new global reference variables if necessary.
+        for (block_id, block) in body.blocks.iter_mut_enumerated() {
+            for inst in block.iter_insts_mut() {
+                match inst {
+                    Inst::Assign(variable, rvalue) => {
+                        update_rvalue(rvalue, &vars_map);
+                        let var_id = variable.as_var_id().unwrap();
+                        let (var_kind, updated_var_id, global_exists) =
+                            vars_map.get_mut(&var_id).unwrap();
+                        if !*global_exists {
+                            *updated_var_id = Some(var_id);
+                            *global_exists = true;
+                        } else {
+                            let new_var_id = body.vars.push_and_get_key(*var_kind);
+                            variable.base = Base::Var(new_var_id);
```

Comment:
Don't forget to check that the projection's array is empty, since

```javascript
let a = { foo: 1 };
a.bar = 2;
a.foo = 3;
```

`a` should resolve to the same def id.